contrib/google.golang.org/grpc/grpc_test.go

```diff
@@ -663,3 +664,111 @@ func TestIgnoredMethods(t *testing.T) {
 		}
 	})
 }
+
+func TestWithMetadataTags(t *testing.T) {
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	rig, err := newRig(true, WithMetadataTags())
+	if err != nil {
+		t.Fatalf("error setting up rig: %s", err)
+	}
+	defer rig.Close()
+
+	ctx := context.Background()
+	ctx = metadata.AppendToOutgoingContext(ctx, "test-key", "test-value")
+	span, ctx := tracer.StartSpanFromContext(ctx, "x", tracer.ServiceName("y"), tracer.ResourceName("z"))
+	rig.client.Ping(ctx, &FixtureRequest{Name: "pass"})
+	span.Finish()
+
+	spans := mt.FinishedSpans()
+
+	var serverSpan mocktracer.Span
+
+	for _, s := range spans {
+		switch s.OperationName() {
+		case "grpc.server":
+			serverSpan = s
+		}
+	}
+
+	span, _ = tracer.SpanFromContext(ctx)
+	assert.NotContains(t, serverSpan.Tags(), tagMetadataPrefix+"x-datadog-trace-id")
+	assert.NotContains(t, serverSpan.Tags(), tagMetadataPrefix+"x-datadog-parent-id")
+	assert.NotContains(t, serverSpan.Tags(), tagMetadataPrefix+"x-datadog-sampling-priority")
+	assert.Equal(t, serverSpan.Tag(tagMetadataPrefix+"test-key"), []string{"test-value"})
```

Comment:
I wonder the same here. Can we not just include this into another already existing test case?